src/coreclr/jit/lsra.cpp

```diff
@@ -8486,18 +8514,16 @@ void LinearScan::resolveEdge(BasicBlock*      fromBlock,
                         tempReg = tempRegFlt;
                 }
 #ifdef TARGET_XARCH
-                else
+                else if (tempRegInt == REG_NA)
```

Comment:
I didn't follow this change. What is this about?

Reply:
Previously we would always use `xchg`/swap. Now we only do that if we weren't able to get a temporary register.